src/components/TodoList/TodoList.jsx

```diff
@@ -1,44 +1,114 @@
 import React from 'react';
+import PropTypes from 'prop-types';
+import classNames from 'classnames';
 import './TodoList.scss';
 
-export const TodoList = () => (
-  <div className="TodoList">
-    <h2>Todos:</h2>
-
-    <div className="TodoList__list-container">
-      <ul className="TodoList__list">
-        <li className="TodoList__item TodoList__item--unchecked">
-          <label>
-            <input type="checkbox" readOnly />
-            <p>delectus aut autem</p>
-          </label>
-
-          <button
-            className="
-              TodoList__user-button
-              TodoList__user-button--selected
-              button
-            "
-            type="button"
-          >
-            User&nbsp;#1
-          </button>
-        </li>
-
-        <li className="TodoList__item TodoList__item--checked">
-          <label>
-            <input type="checkbox" checked readOnly />
-            <p>distinctio vitae autem nihil ut molestias quo</p>
-          </label>
-
-          <button
-            className="TodoList__user-button button"
-            type="button"
-          >
-            User&nbsp;#2
-          </button>
-        </li>
-      </ul>
-    </div>
-  </div>
-);
+export class TodoList extends React.Component {
+  state = {
+    status: 'All',
+    query: '',
+  }
+
+  handleSelect = ({ target }) => {
+    this.setState({ status: target.value });
+  }
+
+  handleChange = ({ target }) => {
+    this.setState({ query: target.value.toLowerCase() });
+  }
+
+  selectButtonToggler = (id) => {
+    this.props.selectButton(id);
+    this.props.setUser(id);
+  }
+
+  render() {
+    const { changeStatus, userId } = this.props;
+    const { status, query } = this.state;
+    let { todos } = this.props;
+
+    switch (status) {
+      case 'Completed':
+        todos = todos.filter(todo => todo.completed);
+        break;
+      case 'Not completed':
+        todos = todos.filter(todo => !todo.completed);
+        break;
+      default:
+        break;
+    }
+
+    if (query) {
+      todos = todos.filter(todo => todo.title.includes(query));
+    }
```

Comment:
let's move all the logic related to filtering todos to App, Todolist will be only responsible for rendering already filtered todos and send query and status to App
you have todos in App state, so you should manipulate them only there

Reply:
I don't know where to insert switch statement(